src/posix/FileSystemPosix.cpp

```diff
@@ -57,7 +57,20 @@ namespace FileSystem {
 #ifdef __APPLE__
 		path += "Library/Application Support/Pioneer";
 #else
-		path += ".pioneer";
+		struct stat info;
+		stat((path + ".pioneer").c_str(), &info);
+		if (S_ISDIR(info.st_mode)) {
+			// Check for legacy pioneer directory.
+			path += ".pioneer";
+		} else {
+			char *data_home = getenv("XDG_DATA_HOME");
+			if (data_home == NULL || strcmp(data_home, "")) {
```

Comment:
`strcmp` returns 0 when two strings are equal, it's a holdover from the C API. As written, this branch will be taken when `data_home` is anything other than an empty string.

Reply:
Yep, this was the sort of thing I was worried about. It's fixed now.